Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -1189,4 +1224,144 @@ extension MergedSQLiteBookmarks {
+                        "(guid, type, parentid, parentName, feedUri, siteUri, pos, title, description,",
+                        "bmkUri, tags, keyword, folderName, queryId, server_modified)",
+                        "SELECT guid, type, parentid, parentName, feedUri, siteUri, pos, title, description,",
+                        "bmkUri, tags, keyword, folderName, queryId,",
+
+                        // This will be fixed up in batches after the initial copy.
+                        "0 AS server_modified",
+                        "FROM \(TableBookmarksLocal) WHERE guid IN",
+                        varlist
+                        ].joined(separator: " ")
+                    _ = change(copySQL, args: args)
+            }
+
+            if err != nil {
+                return false
+            }
+
+            op.bufferValuesToMoveFromLocal
+                .withSubsetsOfSize(BrowserDB.MaxVariableNumber) { guids in
+                    guard err == nil else { return }
+                    let args: Args = guids.map { $0 }
+                    let varlist = BrowserDB.varlist(guids.count)
+
+                    let sqlLocalStructure = "DELETE FROM \(TableBookmarksLocalStructure) WHERE parent IN \(varlist)"
+                    if !change(sqlLocalStructure, args: args) {
+                        return
+                    }
+
+                    let sqlLocal = "DELETE FROM \(TableBookmarksLocal) WHERE guid IN \(varlist)"
+                    if !change(sqlLocal, args: args) {
+                        return
+                    }
+            }
+
+            if err != nil {
+                return false
+            }
+
+            op.modifiedTimes.forEach { (time, guids) in
+                if err != nil { return }
+
+                // This will never be too big: we upload in chunks
+                // smaller than 999!
+                precondition(guids.count < BrowserDB.MaxVariableNumber)
+
+                log.debug("Swizzling server modified time to \(time) for \(guids.count) GUIDs.")
+                let args: Args = guids.map { $0 }
+                let varlist = BrowserDB.varlist(guids.count)
+                let updateSQL = [
+                    "UPDATE \(TableBookmarksBuffer) SET server_modified = \(time)",
+                    "WHERE guid IN",
+                    varlist,
+                    ].joined(separator: " ")
+                _ = change(updateSQL, args: args)
+            }
+
+            if err != nil {
+                return false
+            }
+
+            if let e = deleteStructureForGUIDs([mobileRoot.guid], fromTable: TableBookmarksBufferStructure, connection: conn) {
+                err = e
+                deferred.fill(Maybe(failure: DatabaseError(err: err)))
+                return false
+            }
+
+            if let e = insertStructureIntoTable(TableBookmarksBufferStructure, connection: conn, children: mobileRoot.getChildrenArgs(), maxVars: BrowserDB.MaxVariableNumber) {
```

Comment:
Why did you choose this approach — delete then insert all — rather than appending the uploaded children?

My DB has 102 mobile bookmarks; this means each iOS addition will delete 102 rows and insert 103.